main/build.gradle

```diff
@@ -440,22 +440,24 @@ static def versionNameFromDate() {
 
 class GitCommit {
     static String commitId
-}
 
-/*
- * get the most recent git commit ID
- */
-def gitCommitId() {
-    // TODO: someone with proper groovy knowledge should make this a method at the above class
-    if (GitCommit.commitId == null) {
-        def stdout = new ByteArrayOutputStream()
-        exec {
-            commandLine 'git', 'rev-parse', 'HEAD'
-            standardOutput = stdout
+    /*
+     * get the most recent git commit ID
+     */
+    static String current() {
+        if (commitId == null) {
+            Process p = ['git', 'rev-parse', 'HEAD'].execute()
```

Comment:
It seems challenging to run the `exec` task from within a class method (as it resolves the symbol from that context failing to find class field/method of that name). Refactoring to standard groovy process invocation that is not all that non-idiomatic.